File: Design-Patterns/Mediator/CSharp/Mediator/Employees/CustomerService.cs
using Mediator.Managers;

namespace Mediator.Employees
{
    public class CustomerService : Employee
    {
        public CustomerService(string name, Manager manager) : base(name, manager)
        {
            
        }

        public void Negotiate(Job job)
        {
            // Negotiates with customer.
        }

        public void Deliver(Job job)
        {
            // Delivers job to customer.
        }

        public override bool TakeJob(Job job, bool isForceMode = false)
        {
            Manager.TransferJob(job, this);
            return true;
        }
    }
}

File: Design-Patterns/Mediator/Cpp/Mediator/customer_service.cpp
#include "employee.h"

customer_service::customer_service(std::string name, manager* manager) : employee(name, manager)
{

}

void customer_service::negotiate(job job)
{
	// Negotiates with customer.
}

void customer_service::deliver(job job)
{
	// Delivers job to customer.
}

bool customer_service::take_job(job job, bool is_force_mode)
{
	manager_->transfer_job(job, this);
	return true;
}
